src/internal.c

```diff
@@ -5575,6 +5575,9 @@ void FreeKey(WOLFSSL* ssl, int type, void** pKey)
         #ifdef HAVE_ECC
             case DYNAMIC_TYPE_ECC:
                 wc_ecc_free((ecc_key*)*pKey);
+                #ifdef FP_ECC
+                    wc_ecc_fp_free();
```

Comment:
This is not the correct place for the fixed point cache cleanup. This should only be called when a thread is terminating. If thread local storage is enabled then it needs called for each thread only when it terminates. If thread local storage is not enabled then the call in wolfSSL_cleanup handles it.

Reply:
@dgarske, Thanks, but I don't see a call to wolfSSL_cleanup() from SSL/TLS. What would be the side effect the way I have it?

Reply:
`wolfSSL_Cleanup` -> `wolfCrypt_Cleanup` -> `wc_ecc_fp_free`

Note: Its probably a place in our test tool that is not call `wc_ecc_fp_free` for a thread. You'll need to trace down the call stack to figure out which one.